src/PrefixTree.java

```diff
@@ -0,0 +1,84 @@
+import java.util.ArrayList;
+
+public class PrefixTree {
+
+    private Node root;
+
+    public void setRoot(Node root) {
+        this.root = root;
+    }
+
+    public Node getRoot() {
+        return this.root;
+    }
+
+    private static class Helper{
+        private Node node;
+        private int index;
+
+    }
+
+    public boolean contains(CharSequence word){
+        Helper helper = traverse(word);
+        if(helper.index == -1){
+            System.out.println("Tree contains this word - " + word);
+            return true;
+        }else{
+            System.out.println("Tree doesn't contain this word - " + word);
+            return false;
+        }
+    }
+
+    public boolean add(CharSequence word){
+        Helper helper = traverse(word);
+        Node previousNode = helper.node;
+        if(helper.index == -1){
+            System.out.println("Word has been added already. " + word);
+            return false;
+        }
+        else{
+            for (int i = helper.index; i < word.length(); i++) {
+                if(word.length() == 1){
+                    break;
+                }
+                Edge edge = new Edge();
+                Node node = new Node();
+                node.setValue(word.charAt(i));
+                edge.setReference(node);
+                previousNode.addEdge(edge);
+                previousNode = node;
+            }
+            previousNode.setFinal(true);
+            return true;
+        }
+    }
+
+    private Helper traverse(CharSequence word){
```

Comment:
This looks correct. If you'd like to improve its readabilty further, without resorting to recursion, drop both `i` and `previousNode` from the code and use directly the two fields, returning whatever the result at the end. You'll be able to get rid of the branching this way.